app/models/mixins/supports_feature_mixin.rb

```diff
@@ -96,34 +79,26 @@ def supports_not(feature, reason: nil)
       self.supports_features = supports_features.merge(feature.to_sym => reason.presence || false)
     end
 
-    # query the class if the feature is supported or not
     def supports?(feature)
-      check_supports(feature.to_sym, :instance => self)
+      !unsupported_reason(feature)
     end
 
-    def check_supports(feature, instance:)
-      instance.send(:unsupported).delete(feature)
-
+    # query the class if the feature is supported or not
+    def unsupported_reason(feature, instance: self)
       # undeclared features are not supported
       value = supports_features[feature.to_sym]
-
       if value.respond_to?(:call)
         begin
           # for class level supports, blocks are not evaluated and assumed to be true
           result = instance.instance_eval(&value) unless instance.kind_of?(Class)
-          # if no errors yet but result was an error message
-          # then add the error
-          if !instance.send(:unsupported).key?(feature) && result.kind_of?(String)
-            instance.send(:unsupported_reason_add, feature, result)
-          end
+          result if result.kind_of?(String)
```

Comment:
This is the crux of this pr.

Before:
When asked, clear `unsupported` and then setting it with new value.

After:
return the value and don't use `unsupported`